static/js/containers/ProfileFormContainer.js

```diff
@@ -68,24 +70,41 @@ class ProfileFormContainer extends React.Component {
     }
   };
 
-  updateProfileValidation(props: Object, profile: Profile, validator: Validator|UIValidator): void {
+  updateProfileValidation: Function = (profile: Profile, validator: Validator|UIValidator): void => {
     const username = SETTINGS.username;
-    const { dispatch, profiles, ui } = props;
-    if ( profiles[username].edit && !_.isEmpty(profiles[username].edit.errors) ) {
-      let errors = validator(profile, ui);
-      dispatch(updateProfileValidation(username, errors));
-    }
+    const { dispatch, ui } = this.props;
+    let errors = validator(profile, ui);
+    dispatch(updateProfileValidation(username, errors));
   }
 
-  updateProfile: UpdateProfile = (isEdit, profile, validator) => {
+  updateProfile: UpdateProfile = (isEdit, profile, validator, skipValidation = false) => {
     const { dispatch } = this.props;
     const username = SETTINGS.username;
 
     if (!isEdit) {
       dispatch(startProfileEdit(username));
     }
     dispatch(updateProfile(username, profile));
-    this.updateProfileValidation(this.props, profile, validator);
+    if ( !skipValidation ) {
+      this.updateProfileValidation(profile, validator);
+    }
+  };
+
+  updateValidationVisibility: Function = keySet => {
+    const { dispatch, profiles } = this.props;
+    const username = SETTINGS.username;
```

Comment:
Can you update this to `SETTINGS.user.username` here and below, to avoid collisions with #1615?